src/hole.rs

```diff
@@ -264,6 +257,17 @@ fn deallocate(mut hole: &mut Hole, addr: usize, mut size: usize) {
                 hole = move_helper(hole).next_unwrap(); // start next iteration at next hole
                 continue;
             }
+            _ if hole_addr + hole.size == addr => {
```

Comment:
No big reason, just wanted to keep it close to the other _ case, but I am happy to move it back
Thanks!